src/Phpro/SoapClient/Middleware/WsaMiddleware.php

```diff
@@ -11,6 +11,7 @@
 class WsaMiddleware extends Middleware
 {
     const WSA_ADDRESS_ANONYMOUS = 'http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous';
+    const WSA_ADDRESS2005_ANONYMOUS = 'http://www.w3.org/2005/08/addressing';
```

Comment:
maybe this could/should be renamed to 

```suggestion
    const WSA_ADDRESS2005 = 'http://www.w3.org/2005/08/addressing';
```

since the rule/anonymous part is no longer part of the url

Reply:
You are confusing 2 different settings in here:

There is the WSA namespace, which could be either:

* WSA xmlsoap 2004 : 'http://schemas.xmlsoap.org/ws/2004/08/addressing'
* WSA w3c 2005 : 'http://www.w3.org/2005/08/addressing'

This namespace is used to prefix the added xml elements according to the selected version.
For example:

`xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing"`

Meaning that all `<wsa:x />` elements point to the 2004 xmlsoap version.


Besides the namespace, there is the anonymous Reply-To field.
This is to specify where the webserver needs to send its reply to.
If it is anonymous, it takes current session to send the response to.

This endpoint is different for wsa xmlsoap 2004 and wsa w3c 2005:

* WSA xmlsoap 2004 : 'http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous'
* WSA w3c 2005 : 'http://www.w3.org/2005/08/addressing/anonymous'

More info, see : https://www.w3.org/TR/2005/CR-ws-addr-soap-20050817/#soaphttp

Reply:
thx for your input.

I have separated the 2005-middleware into its own file was a separate test.
hopefully thats what you had in mind and I got you right.